src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -353,15 +338,18 @@
                         (if (:catalog-facts-bash command-processing)
                           (queue/sorted-command-buffer max-enqueued)
                           max-enqueued))
+          q (queue/create-or-open-stockpile (conf/stockpile-dir config))
```

Comment:
I'd suggest grabbing the next-likely-id here, and passing it to the loader below so that we'll be insensitive to the startup time of any other producers/consumers.